### PR TITLE
[Feature Fix] twofactor warning message 

### PR DESCRIPTION
**Pull request by zkraime**
<h2>Purpose:</h2> 
<p>Issue #2385: To update the warning message after authenticating the two-factor addon so that it doesn't continue to say "Please activate your device before continuing"</p>

<h2>Changes:</h2>
<p>If entered the correct verification code, hitting submit now reloads the page and displays the new message</p>
screen shot 2015-05-26 at 3 12 04 pm

<h2>Side effects:</h2>
<p>Don't think there should be any side effects unless reloading the page here leads to undesired behavior elsewhere</p>

### CODE REVIEW

**Comment by zkraime** [comment]
I've been merging my fix branches back into my develop branch before pushing - I realize now that I should not have been doing this

**Comment by lbanner** [comment]
:+1:

File: website/templates/public/pages/getting_started.mako [hunk 1]
```diff
@@ -4,7 +4,7 @@
     <div class="row"  href="#start">
         <div class="col-sm-3 nav-list-spy">
             <div data-spy="affix" class="gs-sidebar hidden-print hidden-xs panel panel-default" role="complementary">
-                <ul class="nav nav-stacked nav-pills gs-sidenav" style="min-width: 210px">
+                <ul class="nav nav-stacked nav-pills gs-sidenav" style=" min-width: 210px">
```
Comment (HarryRybacki) [inline website/templates/public/pages/getting_started.mako]: Extra space in your styling.

File: website/addons/twofactor/views.py [hunk 2]
```diff
@@ -20,7 +21,8 @@ def user_settings(user_addon, *args, **kwargs):
     if user_addon.verify_code(code):
         user_addon.is_confirmed = True
         user_addon.save()
-        return {'message': 'Successfully verified two-factor authentication.'}, http.OK
+        push_status_message('Successfully verified two-factor authentication!')
```
Comment (samanehsan) [inline website/addons/twofactor/views.py]: This alert should be green instead of the default warning-yellow since it indicates success.

**Comment by samanehsan** [comment]
This pull request has some css changes to `getting_started.mako` -- were they meant to fix a different issue?

File: website/addons/twofactor/views.py (hunk 2 above)
Comment (samanehsan) [inline website/addons/twofactor/views.py]: Also, minor nit-pick but perhaps replace the exclamation mark with a period?

Comment (zkraime) [inline website/addons/twofactor/views.py]: @samanehsan, I'll change this - thanks. 
As for the css changes to getting_started.mako, originally I had inserted some inline css, so I removed that but I noticed that there was some other inline css that originally there that I could refactor out. Sorry for not specifying this in the PR

File: website/templates/public/pages/getting_started.mako (hunk 1 above)
Comment (zkraime) [inline website/templates/public/pages/getting_started.mako]: I thought I got rid of this, my fault. I'll change it

File: website/addons/twofactor/views.py (hunk 2 above)
Comment (samanehsan) [inline website/addons/twofactor/views.py]: Great, thanks for the clarification!

**Comment by samanehsan** [comment]
Could you remove your changes to `getting_started.mako` from this PR? This looks different from the fix in #2864 and that may get confusing.

**Comment by zkraime** [comment]
Yes right away

File: website/templates/public/pages/getting_started.mako [hunk 3]
```diff
@@ -110,7 +110,7 @@
         </div>
 
 
-        <div  id="structure" class="row col-sm-12 col-md-12" style="padding-top: 40px;">
+        <div  id="structure" class="row col-sm-12 col-md-12" style="padding-top: 40px">
```
Comment (samanehsan) [inline website/templates/public/pages/getting_started.mako]: Can you add the semicolons back in/make sure there's no diff with this file? It's best to include only the changes relevant to the issue that you are solving.

**Comment by samanehsan** [comment]
Just one small change and this should be good to go! :octopus:

File: website/addons/twofactor/views.py [hunk 4]
```diff
@@ -20,7 +21,8 @@ def user_settings(user_addon, *args, **kwargs):
     if user_addon.verify_code(code):
         user_addon.is_confirmed = True
         user_addon.save()
-        return {'message': 'Successfully verified two-factor authentication.'}, http.OK
+        push_status_message('Successfully verified two-factor authentication.', kind='success')
+        return http.OK
```
Comment (sloria) [inline website/addons/twofactor/views.py]: This is incorrect. Views must return either response data or a tuple where the first element is the response data, 2nd is the status code (optional), 3rd is headers (optional).

**Comment by sloria** [comment]
This approach can be improved: the `user_settings` endpoint is an JSON endpoint and is part of the API. As such, it should be stateless--it should not rely, for example, on session state (status messages are saved on the session).

So instead of pushing a status message, you can instead use the JSON response body returned by the view and display the message using Javascript--I see that there is already code in the ViewModel for displaying messages (the message and messageClass observables). This obviates the need for a page refresh.

**Comment by zkraime** [comment]
@sloria Yes, thanks for the clarification

**Comment by zkraime** [comment]
@sloria Should I display the JSON response as a slide-down from the top of the screen or the page? 
Also, can I have a suggestion as how to remove the status message using Javascript without a reload?

**Pull request closed without merging**